frontend/apps/reader/modules/readertoc.lua

```diff
@@ -489,8 +510,31 @@ function ReaderToc:onShowToc()
         for _,v in ipairs(self.toc) do
             v.text = self.toc_indent:rep(v.depth-1)..self:cleanUpTocTitle(v.title)
             v.mandatory = v.page
+            local flow = self.ui.document:getPageFlow(v.page)
+            local orig_flow = flow
             if v.orig_page then -- bogus page fixed: show original page number
-                v.mandatory = T("(%1) %2", v.orig_page, v.page)
+                -- This is an ugly piece of code, which can result in an ugly TOC,
+                -- but it shouldn't be needed very often
```

Comment:
Good to say it's ugly, but please also say why it's rare and what this intends to do/show :)
I'd also like this wrapped in a `if  ...:hasHiddenFlows() then`.